pkg/reconciler/ingress/lister.go

```diff
@@ -66,14 +66,14 @@ func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1al
 }
 
 func (l *gatewayPodTargetLister) getIngressUrls(ing *v1alpha1.Ingress, gatewayIps []string) ([]status.ProbeTarget, error) {
-	var targets []status.ProbeTarget
 	localDomainName := network.GetClusterDomainName()
 	ips := sets.NewString()
 
 	for _, ip := range gatewayIps {
 		ips.Insert(ip)
 	}
 
+	targets := make([]status.ProbeTarget, 0, len(ing.Spec.Rules))
```

Comment:
An internal rule generates one target, but an external one generates 2 targets.
The array will grow as needed but maybe we should leave this one as before to avoid confusion?

Reply:
:man_shrugging: we can also add a comment stating that. Why waste precious CPU cycles?